src/common/registry.rs

```diff
@@ -101,22 +75,28 @@ impl TryFrom<ResourceName> for ResourceKey {
 
 pub struct Dependency(pub ResourceKey, pub Resource);
 
+/// Fn that returns a `BoardType`, `Arc<Mutex<dyn Board>>`
```

Comment:
I'm not too sure these comments add a whole lot. If anything we should change names like `BoardType` to `ThreadSafeBoard` instead of inserting comments like these, but that would be too much for this PR.

Reply:
yeah, I think there's probably a better way to represent these types, whether its by renaming or something else. I'll do some research in the meantime on other options.